Sprint-1/implement/sum.test.js

```diff
@@ -2,18 +2,62 @@
 
 In this kata, you will need to implement a function that sums the numerical elements of an array
 
+
 E.g. sum([10, 20, 30]), target output: 60
 E.g. sum(['hey', 10, 'hi', 60, 10]), target output: 80 (ignore any non-numerical elements)
 */
 
 const sum = require("./sum.js");
 
+
+// Test for an empty array
+test("Given an empty array, return 0", () => {
+  expect(sum([])).toBe(0); // Expect 0
+});
+
+// Test for an array with only numbers
+test("Given an array of numbers, return their sum", () => {
+  expect(sum([10, 20, 30])).toBe(60);
+});
+
+// Test for an array with just one number
+test("Given an array with numbers and strings, sum only the numbers", () => {
+  expect(sum(["a", 10, "b", "c", "d"])).toBe(10);
+});
+
+// Test for an array with negative numbers
+test("Given an array with numbers and strings, sum only the numbers", () => {
+  expect(sum(["a", 10, "b", "c", "d"])).toBe(10);
+});
+
+// Test for an array with only non-numeric values
+test("array containing negative numbers, return the correct total sum", () => {
+  expect(sum(["hello", null, undefined, "world"])).toBe(0);
+});
+
+// Test for array with decimal/float numbers
+test("array with decimal/float numbers, return the correct total sum", () => {
+  expect(sum([3.5, 4.5, 5.5])).toBe(13.5);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?